### DISCUSSION TITLE
Discussion meeting 02-07-2024

### DISCUSSION THREAD

**Discussion by madelondohmen** [minvws/nl-kat-coordination#3173] [Discussion Meetings]
* https://github.com/minvws/nl-kat-coordination/pull/2615
We have partial information, which does not yet fit into our database. Doesn't fit in IPPort's database model either.
It exposes some open points in our application.

  - Options:
    - Introduce a new model (IPPort without protocol) 
    - Making InternetDBBoefje do something without port information, but then some of the functionality would be lost
    - Take a guess on which port is TCP and UDP, but this is not completely secure
    - Scan yourself, but then it is no longer L1 clearance
    - Getting Shodan to scan it

  - What we need to fix in general in OpenKAT:
     - What do you do when a boefje is disabled?
     - How sure are we that the information is true? Resolve conflicting information.

  - Proposal:
Remove IPPort from yielded objects. Change in the model is therefore not needed, making the PR smaller. Additionally, include a warning in the description of the boefje that potentially outdated information remains when the boefje is disabled.

* https://github.com/minvws/nl-kat-coordination/pull/3114
You wouldn't want to see JSON output, but maybe we should make it configurable. After all, you don't want to log twice. What is convenient to do? 

  - Proposal:
Feature flag to have JSON output in structured logging or not.

* https://github.com/minvws/nl-kat-coordination/pull/2986
When a component is built, it is now built and tested on a random page. Then all of this is thrown away, which is a waste. Might we want to add a style page where we set everything up?

  - Options:
    - Add theme within Manon
    - Storybook (external party)
    - Put in Rocky itself
    - Make Manon documentation site importable

  - Proposal:
Import documentation from Manon within OpenKAT. Combine with our own OpenKAT style and offer there to our users.

* https://github.com/minvws/nl-kat-coordination/pull/2174
  - In design, it is now established like this:
    - You choose a selection on the current page and continue with it.
    - Or you create a query (a live set) that you use to select data.

  - Idea: 
If you have a selection and you press the next button, maybe there should be a warning that you lost your selection.

  - Proposal:
It is not clear to everyone how this is going to work for the users. The discussions about this topic have already been closed, so we're going to look up the conclusions and the designs and then organise a meeting to inform everyone about these conclusions.

**Comment by zcrt** [comment]
> How sure are we that the information is true? Resolve conflicting information.

Isn't octopoes designed to address this? 
https://docs.openkat.nl/developer_documentation/octopoes_research.html#contradicting-information. Confidence scores seam a reasonable approach.
